mountpoint-s3/src/prefetch/part.rs

```diff
@@ -11,17 +15,28 @@ pub struct Part {
     key: String,
     offset: u64,
     bytes: Bytes,
+    checksum: ChecksumCrc,
 }
 
 impl Part {
     pub fn new(key: &str, offset: u64, bytes: Bytes) -> Self {
+        let checksum = checksums_crc32c(&bytes, None);
         Self {
             key: key.to_owned(),
             offset,
             bytes,
+            checksum,
         }
     }
 
+    pub fn checksum(&self) -> ChecksumCrc {
+        self.checksum
+    }
+
+    pub fn bytes_ref(&self) -> &[u8] {
+        self.bytes.as_ref()
+    }
```

Comment:
This method shouldn't exist, it defeats the purpose of this interface (which prevents you from retrieving the bytes unless you know the key/offset).